extension/httpforwarderextension/extension.go

```diff
@@ -47,7 +49,9 @@ func (h *httpForwarder) Start(ctx context.Context, host component.Host) error {
 		return fmt.Errorf("failed to create HTTP Client: %w", err)
 	}
 
+	h.shutdownWG.Add(1)
 	go func() {
+		defer h.shutdownWG.Done()
```

Comment:
Nit: I don't think we need a WG here, it can be a channel instead if we need this synronization

Reply:
Sure, I'm just curious if there is a specific reason for that: is it cheaper or faster? I tend to use WG if there is no need to flow data between concurrent code. That said, I've seen plenty of code using channels for this kind of sync. At first the disadvantage that I see for a channel is that it requires a call to `make(chan struct{})` but I don't know if the compiler optimizes that or not.

Reply:
Sure, let's keep it. Doesn't matter that much. I wrote after misreading where the Wait is called.